torch_xla_py/xla_multiprocessing.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import division
+from __future__ import print_function
+
+import contextlib
+import os
+import socket
+import sys
+import torch.multiprocessing
+import torch_xla
+import torch_xla_py.xla_env_vars as xenv
+import torch_xla_py.xla_model as xm
+import traceback
+
+
+def _find_free_tcp_port():
+  with contextlib.closing(socket.socket(socket.AF_INET,
+                                        socket.SOCK_STREAM)) as s:
+    s.bind(('', 0))
+    return s.getsockname()[1]
+
+
+def _is_tpu_config():
+  for env in [xenv.TPU_CONFIG, xenv.LOCAL_WORKER]:
+    if os.environ.get(env, None) is not None:
+      return True
+  return False
+
+
+def _parse_tpu_config(config):
+  # XRT_TPU_CONFIG='tpu_worker;0;ismz9:25822'
+  parsed = []
+  for worker in config.split('|'):
+    parts = worker.split(';')
+    if len(parts) != 3:
+      raise ValueError('Bad worker syntax: {}'.format(worker))
+    parsed.append((parts[0], int(parts[1]), parts[2]))
+  return parsed
+
+
+def _get_devices_per_worker():
+  return 8
+
+
+def _get_multiprocessing_device():
+  return os.environ.get(xenv.MP_DEVICE, None)
+
+
+def _get_local_worker_index():
+  worker = os.environ.get(xenv.LOCAL_WORKER, None)
+  return int(worker.split(':')[1]) if worker is not None else 0
+
+
+def _local_index_to_global(index):
+  return _get_local_worker_index() * _get_devices_per_worker() + index
+
+
+def _get_world_size():
+  wsize = int(os.environ.get(xenv.WORLD_SIZE, '1'))
```

Comment:
We can use `xm.xrt_world_size()` instead?

Reply:
> We can use `xm.xrt_world_size()` instead?

We need to be careful in calling into `xm` because we do not know if calls into that code, triggers XLA library initialization (which they should not at some stages in the multiprocessing code).